### PR TITLE
[CLI-664] Better error msg when logging to non existent local instance

### PR DESCRIPTION
**Pull request by farizrahman4u**
https://wandb.atlassian.net/browse/CLI-664

Description
-----------
Better error message when logging to non existent local instance

Testing
-------

Manually

### CODE REVIEW

**Review by vanpelt — COMMENTED** [review]
Hmm, this is definitely better than what we have, but not ideal.  Ideally we would not enter into a retry loop when we can't connect to the local instance on boot.  The retrying of invalid JSON also seems suspect.  I can't think of a scenario where we don't return JSON with a 200 response and actually want to retry...

File: wandb/sdk/internal/internal_api.py [hunk 1]
```diff
@@ -98,13 +111,20 @@ def __init__(
...
+                check_retry_fn=util.no_retry_auth,
+                retryable_exceptions=(RetryError, requests.RequestException,),
+                retry_warning=local_instance_warning,
+                retry_callback=retry_callback,
+            )
+        except Exception as e:
+            if local_instance_warning:
+                raise Exception(local_instance_warning)
+            else:
+                raise e
```
Comment (vanpelt) [inline wandb/sdk/internal/internal_api.py:127]: What is the `try` -> `except` for?  Won't the retrier keep retrying for hours and just print a warning?

File: wandb/sdk/internal/internal_api.py [hunk 2]
```diff
@@ -69,12 +69,25 @@ def __init__(
...
         self.default_settings.update(default_settings or {})
         self.retry_uploads = 10
         self._settings = Settings(
             load_settings=load_settings, root_dir=self.default_settings.get("root_dir")
         )
+
+        base_url = self.settings("base_url")
+        if not (
+            base_url.endswith("api.wandb.ai") or base_url.endswith("api.wandb.ai/")
+        ):
```
Comment (vanpelt) [inline wandb/sdk/internal/internal_api.py:83]: `not "api.wandb.ai" in base_url`

**Review by vanpelt — CHANGES_REQUESTED** [review]

File: wandb/sdk/internal/internal_api.py (hunk 1 above)
Comment (farizrahman4u) [reply wandb/sdk/internal/internal_api.py:127]: Its for the JSONDecodeError (see  jira ticket)

Comment (vanpelt) [reply wandb/sdk/internal/internal_api.py:127]: The recent retry fixes should ensure we don't raise exceptions when content isn't JSON encoded so the retry logic will retry it forever and this except case will only be reached after we stop retrying...

**Pull request closed without merging**